include/glow/Graph/Graph.h

```diff
@@ -627,6 +627,55 @@ class Function final : public Named {
   ARITHMETIC_FUN_DECL(Pow);
 #undef ARITHMETIC_FUN_DECL
 
+  std::vector<NodeValue>
+  broadcastInputs(int axis, const llvm::ArrayRef<NodeValue> inputs);
+
+  template <class T, class U>
+  using enable_if_same_t = std::enable_if<std::is_same<T, U>::value, U>;
+#define BROADCAST_FUNC_T(NODE_NAME_)                                           \
```

Comment:
I like the `enable_if` trick here (IIRC you did something similar in the optimizer), but I'd really prefer not to hide it in a macro.  It makes the definitions a bit shorter but relatively more confusing.  (When I looked at the optimizer it took me some time to mentally unwind the macro).

Reply:
Yep. :)
So replace with full version?
i.e?
template <class T, class... Args>
typename enable_if_same_t<T, MulNode>::type

What about?
BROADCAST_FUNC_COMMON_CODE

Reply:
Sgtm!  I'm cool with BROADCAST_FUNC_COMMON_CODE since it's kind of boilerplate-y asserts rather than tricksy template magic :-).